types/instance.d.ts

```diff
@@ -190,7 +190,7 @@ export interface FastifyInstance<
   listen(port: number | string, address?: string, backlog?: number): Promise<string>;
 
   ready(): FastifyInstance<RawServer, RawRequest, RawReply, Logger, TypeProvider> & PromiseLike<undefined>;
-  ready(readyListener: (err: Error) => void): FastifyInstance<RawServer, RawRequest, RawReply, Logger, TypeProvider>;
+  ready(readyListener: (err?: Error) => void): FastifyInstance<RawServer, RawRequest, RawReply, Logger, TypeProvider>;
```

Comment:
```suggestion
  ready(readyListener: (err: Error | null) => void): FastifyInstance<RawServer, RawRequest, RawReply, Logger, TypeProvider>;
```

Reply:
@climba03003 

> error should be null by default. If it doesn't, better to fix it inside avvio.
https://github.com/fastify/avvio/blob/master/boot.js#L447-L448

I agree with you. This `null` behavior is a standard way of Node.js.

I commit your suggestion, but if the difference between the current `undefined` behavior and `null` type definition has any problem, please let me know.

Also, I'll look into avvio implementation.